src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchRestTemplate.java

```diff
@@ -1474,121 +1409,90 @@ public void refresh(String indexName) {
 		}
 	}
 
-	@Nullable
-	private ElasticsearchPersistentEntity<?> getPersistentEntity(@Nullable Class<?> clazz) {
-		return clazz != null ? elasticsearchConverter.getMappingContext().getPersistentEntity(clazz) : null;
+	protected ResultsMapper getResultsMapper() {
+		return resultsMapper;
 	}
 
-	@Override
-	public ElasticsearchPersistentEntity getPersistentEntityFor(Class clazz) {
-		Assert.isTrue(clazz.isAnnotationPresent(Document.class), "Unable to identify index name. " + clazz.getSimpleName()
-				+ " is not a Document. Make sure the document class is annotated with @Document(indexName=\"foo\")");
-		return elasticsearchConverter.getMappingContext().getRequiredPersistentEntity(clazz);
+	@Deprecated
+	public static String readFileFromClasspath(String url) {
+		return ResourceUtil.readFileFromClasspath(url);
 	}
 
-	private String getPersistentEntityId(Object entity) {
-
-		ElasticsearchPersistentEntity<?> persistentEntity = getPersistentEntityFor(entity.getClass());
-		Object identifier = persistentEntity.getIdentifierAccessor(entity).getIdentifier();
+	public SearchResponse suggest(SuggestBuilder suggestion, String... indices) {
 
-		if (identifier != null) {
-			return identifier.toString();
-		}
+		Assert.notNull(indices, "No index defined for query");
+		SearchRequest searchRequest = new SearchRequest(indices);
+		searchRequest.source().suggest(suggestion);
 
-		return null;
+		return search(searchRequest);
 	}
 
-	private void setPersistentEntityId(Object entity, String id) {
-
-		ElasticsearchPersistentEntity<?> persistentEntity = getPersistentEntityFor(entity.getClass());
-		ElasticsearchPersistentProperty idProperty = persistentEntity.getIdProperty();
-
-		// Only deal with text because ES generated Ids are strings !
-
-		if (idProperty != null && idProperty.getType().isAssignableFrom(String.class)) {
-			persistentEntity.getPropertyAccessor(entity).setProperty(idProperty, id);
-		}
+	public SearchResponse suggest(SuggestBuilder suggestion, Class<?> clazz) {
+		return suggest(suggestion, operations.determineIndexName(clazz));
 	}
 
-	private void setPersistentEntityIndexAndType(Query query, Class clazz) {
-		if (query.getIndices().isEmpty()) {
-			query.addIndices(retrieveIndexNameFromPersistentEntity(clazz));
-		}
-		if (query.getTypes().isEmpty()) {
-			query.addTypes(retrieveTypeFromPersistentEntity(clazz));
+	private SearchResponse search(SearchRequest request) {
```

Comment:
These methods are really handy and reduce the code a lot. Should I rebase this PR or should I open a new one? I could also add an issue for that, if needed.

Reply:
There were quite heavy changes in the internal code structure in DATAES-629 to DATAES-634 especially with the introduction of the `RequestFactory` and the restructuring of the API. I'm not sure if rebasing will really work here

Reply:
I'll apply the changes that makes sense manually and could force-push to that branch. But if you think it's better to close this PR and open a new one, then it's also good for me.

Reply:
I personally would create a new PR with a new issue

Reply:
and please notice that with DATAES-672 I am just changing the return types of the `SearchOperations` methods from `Page<T>` to `Page<SearchHit<T>>`, same for lists etc. I have it change in my branch for the Template implementations, but I have to fix the Repository implementations to be able to work with that.